caf-logging-tomcat-juli/src/main/java/com/github/cafapi/logging/tomcat/juli/CAFTomcatConsoleAppender.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2017-2020 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafapi.logging.tomcat.juli;
+
+import org.apache.juli.logging.ch.qos.logback.classic.spi.LoggingEvent;
+import org.apache.juli.logging.ch.qos.logback.core.ConsoleAppender;
+
+import com.github.cafapi.logging.common.ProcessAndThreadIdProvider;
+
+public final class CAFTomcatConsoleAppender extends ConsoleAppender<LoggingEvent>
```

Comment:
I notice that even though this looks like a normal Logback `ConsoleAppender` that it is actually inheriting from `org.apache.juli.logging.ch.qos.logback.core.ConsoleAppender` rather than `ch.qos.logback.core.ConsoleAppender`.  Is this some sort of fork of Logback that is being used here?  I don't think I understand what's going on.

Reply:
As per the documentation https://github.com/tomcat-slf4j-logback/tomcat-slf4j-logback#configuration, logback.xml should be written with renamed packages from tomcat9-slf4j-logback. 
The ConsoleAppender should be from org.apache.juli.logging.ch.qos.logback.core.ConsoleAppender rather than ch.qos.logback.core.ConsoleAppender. 
Since the relocations happened https://github.com/tomcat-slf4j-logback/tomcat9-slf4j-logback/blob/master/pom.xml#L154 , the CAFConsoleAppender configuration should not be applicable here. 
The configuration file property also changed from  logback.configurationFile to  juli-logback.configurationFile.
When the CAFConsoleAppender is used, The method ch.qos.logback.classic.util.ContextInitializer expects the CONFIG_FILE_PROPERTY value is logback.configurationFile. 
But that is now changed to juli-logback.configurationFile.

Exception stack

0:18:46.743 documentid-service> 10:18:46,734 |-ERROR in ch.qos.logback.core.joran.action.AppenderAction - Could not create an Appender of type [com.github.cafapi.logging.logback.CAFConsoleAppender]. ch.qos.logback.core.util.IncompatibleClassException
10:18:46.743 documentid-service> 	at ch.qos.logback.core.util.IncompatibleClassException
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.core.util.OptionHelper.instantiateByClassNameAndParameter(OptionHelper.java:58)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.core.util.OptionHelper.instantiateByClassName(OptionHelper.java:45)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.core.util.OptionHelper.instantiateByClassName(OptionHelper.java:34)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.core.joran.action.AppenderAction.begin(AppenderAction.java:52)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.core.joran.spi.Interpreter.callBeginAction(Interpreter.java:269)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.core.joran.spi.Interpreter.startElement(Interpreter.java:145)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.core.joran.spi.Interpreter.startElement(Interpreter.java:128)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.core.joran.spi.EventPlayer.play(EventPlayer.java:50)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.core.joran.GenericConfigurator.doConfigure(GenericConfigurator.java:165)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.core.joran.GenericConfigurator.doConfigure(GenericConfigurator.java:152)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.core.joran.GenericConfigurator.doConfigure(GenericConfigurator.java:110)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.core.joran.GenericConfigurator.doConfigure(GenericConfigurator.java:53)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.classic.util.ContextInitializer.configureByResource(ContextInitializer.java:75)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.classic.util.ContextInitializer.autoConfig(ContextInitializer.java:150)
10:18:46.743 documentid-service> 	at 	at org.slf4j.impl.StaticLoggerBinder.init(StaticLoggerBinder.java:84)
10:18:46.743 documentid-service> 	at 	at org.slf4j.impl.StaticLoggerBinder.<clinit>(StaticLoggerBinder.java:55)
10:18:46.743 documentid-service> 	at 	at org.slf4j.LoggerFactory.bind(LoggerFactory.java:150)
10:18:46.743 documentid-service> 	at 	at org.slf4j.LoggerFactory.performInitialization(LoggerFactory.java:124)
10:18:46.743 documentid-service> 	at 	at org.slf4j.LoggerFactory.getILoggerFactory(LoggerFactory.java:412)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.classic.util.StatusViaSLF4JLoggerFactory.addStatus(StatusViaSLF4JLoggerFactory.java:32)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.classic.util.StatusViaSLF4JLoggerFactory.addInfo(StatusViaSLF4JLoggerFactory.java:20)
10:18:46.743 documentid-service> 	at 	at ch.qos.logback.classic.servlet.LogbackServletContainerInitializer.onStartup(LogbackServletContainerInitializer.java:32)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5135)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:717)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:690)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:705)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1133)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1867)
10:18:46.743 documentid-service> 	at 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
10:18:46.743 documentid-service> 	at 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
10:18:46.743 documentid-service> 	at 	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
10:18:46.743 documentid-service> 	at 	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:112)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.startup.HostConfig.deployDirectories(HostConfig.java:1045)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:429)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1576)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:309)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:366)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:936)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:841)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1384)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1374)
10:18:46.743 documentid-service> 	at 	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
10:18:46.743 documentid-service> 	at 	at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
10:18:46.743 documentid-service> 	at 	at java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:134)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:909)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:262)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:421)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:932)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.startup.Catalina.start(Catalina.java:633)
10:18:46.743 documentid-service> 	at 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
10:18:46.743 documentid-service> 	at 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
10:18:46.743 documentid-service> 	at 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
10:18:46.743 documentid-service> 	at 	at java.lang.reflect.Method.invoke(Method.java:498)
10:18:46.743 documentid-service> 	at 	at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:344)